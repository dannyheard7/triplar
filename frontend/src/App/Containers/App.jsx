import React from "react";
import {Route, Router, Switch} from "react-router-dom";
import {DragDropContext} from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import {connect} from "react-redux";
import ReactGA from 'react-ga';
import {Helmet} from "react-helmet";

import {history} from "../../store";
import {trackingId} from "../utils/constants";

import LoggedInContainer from "App/Containers/LoggedInContainer";
import HeaderAuthenticated from "App/Components/HeaderAuthenticated";
import HeaderUnauthenticated from "App/Components/HeaderUnauthenticated";
import Footer from "App/Components/Footer";
import LoginContainer from "Auth/Containers/LoginContainer";
import RegistrationContainer from "Auth/Containers/RegistrationContainer";
import TripCreateContainer from "Trips/Components/TripCreate";
import TripEditContainer from "../../Trips/Components/TripEdit";
import TripDelete from "../../Trips/Components/TripDelete";
import PlaceDetail from "../../Places/Components/PlaceDetail";
import {getCookie} from "../utils/utils";
import {tokenRefresh} from "../../Auth/utils/actions";
import TripDetail from "../../Trips/Components/TripDetail";
import TripList from "../../Trips/Components/TripList";
import ItinerariesOverview from "../../Itinerary/Components/ItinerariesOverview";
import LocationItinerary from "../../Itinerary/Components/LocationItinerary";

import 'App/styles/app.css';

export class App extends React.Component {
    componentDidMount() {
        if(!this.props.auth.successful) {
            const loginToken = getCookie('userToken');

            if (loginToken) {
                this.props.dispatch(tokenRefresh(loginToken));
            }
        }

        this.initializeReactGA();
    }

    initializeReactGA() {
        ReactGA.initialize('UA-130249377-1');
        ReactGA.pageview("/");
        history.listen(location => ReactGA.pageview(location.pathname));
    }

    render() {
        if(this.props.auth.successful) {
            return (
                <Router history={history}>
                    <LoggedInContainer token={this.props.auth.token}>
                        <Helmet>
                            <title>Triplar</title>
                        </Helmet>
                        <div className="wrapper">
                            <HeaderAuthenticated/>
                            <div className="content">
                                <Route path="/(|trips)" exact component={TripList}/>
                                <Route path="/(|trips)" exact component={TripCreateContainer}/>

                                <Route path="/trips/:id" component={TripDetail}/>
                                <Route path="/trips/:id/edit" component={TripEditContainer}/>
                                <Route path="/trips/:id/delete" component={TripDelete} />

                                <Route path="/trips/:id" component={ItinerariesOverview}/>

                                <Route path="/trips/:tripId/itinerary/:itineraryId" component={LocationItinerary}/>

                                <Route path="*/place/:placeId" component={PlaceDetail}/>
                            </div>
                            <Footer/>
                        </div>
                    </LoggedInContainer>
                </Router>
            );
        } else {
            return (
                <Router history={history}>
                    <div className="wrapper">
                        <Helmet>
                            <title>Triplar</title>
                        </Helmet>
                        <HeaderUnauthenticated/>
                        <div className="content">
                            <Switch>
                                <Route path="/register" exact component={RegistrationContainer}/>
                                <Route path="/" component={LoginContainer}/>
                            </Switch>
                        </div>
                        <Footer/>
                    </div>
                </Router>
            );
        }
    }
}

const mapStateToProps = state => {
    if(state.auth.successful) {
        return {
            auth: {...state.auth, token: getCookie('userToken')}
        }
    } else {
        return {auth: state.auth}
    }
};

const connectedApp = connect(mapStateToProps)(App);

export default DragDropContext(HTML5Backend)(connectedApp)