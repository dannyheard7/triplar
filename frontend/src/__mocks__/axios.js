export default {
    get: jest.fn(() => Promise.resolve({ data: {} })),
    post: jest.fn(() => Promise.resolve({ data: {} })),
    patch: jest.fn(() => Promise.resolve({ data: {} })),
    delete: jest.fn(() => Promise.resolve({ data: {} })),
    defaults: jest.fn(),
    create: jest.fn(),
};