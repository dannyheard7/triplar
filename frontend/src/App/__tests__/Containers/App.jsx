import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import axios from "axios";

const renderer = new ShallowRenderer();
const faker = require('faker');

import App from "App/Containers/App";
import {shallow} from "enzyme/build/index";
import {Redirect} from "react-router-dom";

jest.unmock('axios');
describe('<App />', () => {

    describe('when authenticated', () => {
        const user = {'email': faker.internet.email()};

        const props = {
            isAuthenticated: true,
            user: user,
        };

        test.skip('redirects to trips/', () => {
            const container = shallow(<App {...props}/>);
            expect(container.find(Redirect)).toHaveLength(1);
        });
    });

     describe('when not authenticated', () => {
        const props = {
            isAuthenticated: false,
        };

        test('renders correctly', () => {
            const result = renderer.render(<App {...props} />);
            expect(result).toMatchSnapshot();
        });
    });


});