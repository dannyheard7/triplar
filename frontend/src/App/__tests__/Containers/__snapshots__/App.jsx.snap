// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> when authenticated renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <LoggedInContainer>
    <div
      className="wrapper"
    >
      <withRouter(Connect(HeaderAuthenticated)) />
      <div
        className="content"
      >
        <Route
          component={[Function]}
          exact={true}
          path="/(|trips)"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/(|trips)"
        />
        <Route
          component={[Function]}
          path="/trips/:id"
        />
        <Route
          component={[Function]}
          path="/trips/:id/edit"
        />
        <Route
          component={[Function]}
          path="/trips/:id/delete"
        />
        <Route
          component={[Function]}
          path="/trips/:id"
        />
        <Route
          component={[Function]}
          path="/trips/:tripId/place/:placeId"
        />
      </div>
      <Footer />
    </div>
  </LoggedInContainer>
</Router>
`;

exports[`<App /> when not authenticated renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    className="wrapper"
  >
    <HeaderUnauthenticated />
    <div
      className="content"
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/register"
        />
        <Route
          component={[Function]}
          path="/"
        />
      </Switch>
    </div>
    <Footer />
  </div>
</Router>
`;
